nifi-nar-bundles/nifi-atlas-bundle/nifi-atlas-reporting-task/src/main/java/org/apache/nifi/atlas/provenance/lineage/SimpleFlowPathLineage.java

```diff
@@ -0,0 +1,205 @@
+        }
+
+    }
+
+    /**
+     * Create a flow_path entity corresponding to the target RemoteGroupPort when a SEND/RECEIVE event are received.
+     * Because such entity can not be created in advance while analyzing flow statically,
+     * as ReportingTask can not determine whether a component id is a RemoteGroupPort,
+     * since connectionStatus is the only available information in ReportingContext.
+     * ConnectionStatus only knows component id, component type is unknown.
+     * For example, there is no difference to tell if a connected component is a funnel or a RemoteGroupPort.
+     */
+    private void processRemotePortEvent(AnalysisContext analysisContext, NiFiFlow nifiFlow, ProvenanceEventRecord event, DataSetRefs analyzedRefs) {
+
+        final boolean isRemoteInputPort = "Remote Input Port".equals(event.getComponentType());
+
+        // Create a RemoteInputPort Process.
+        // event.getComponentId returns UUID for RemoteGroupPort as a client of S2S, and it's different from a remote port UUID (portDataSetid).
+        // See NIFI-4571 for detail.
+        final Referenceable remotePortDataSet = isRemoteInputPort ? analyzedRefs.getOutputs().iterator().next() :  analyzedRefs.getInputs().iterator().next();
+        final String portProcessId = event.getComponentId();
+
+        final NiFiFlowPath remotePortProcess = new NiFiFlowPath(portProcessId);
+        remotePortProcess.setName(event.getComponentType());
+        remotePortProcess.addProcessor(portProcessId);
+
+        // For RemoteInputPort, need to find the previous component connected to this port,
+        // which passed this particular FlowFile.
+        // That is only possible by calling lineage API.
+        if (isRemoteInputPort) {
+            final ProvenanceEventRecord previousEvent = findPreviousProvenanceEvent(analysisContext, event);
+            if (previousEvent == null) {
+                logger.warn("Previous event was not found: {}", new Object[]{event});
+                return;
+            }
+
+            // Set groupId from incoming connection if available.
+            final List<ConnectionStatus> incomingRelationShips = nifiFlow.getIncomingRelationShips(portProcessId);
+            if (incomingRelationShips == null || incomingRelationShips.isEmpty()) {
+                logger.warn("Incoming relationship was not found: {}", new Object[]{event});
+                return;
+            }
+
+            final ConnectionStatus connection = incomingRelationShips.get(0);
+            remotePortProcess.setGroupId(connection.getGroupId());
+
+            final Referenceable remotePortProcessRef = toReferenceable(remotePortProcess, nifiFlow);
+            createEntity(remotePortProcessRef);
+
+            // Create a queue.
+            Referenceable queueFromStaticFlowPathToRemotePortProcess = new Referenceable(TYPE_NIFI_QUEUE);
+            queueFromStaticFlowPathToRemotePortProcess.set(ATTR_NAME, "queue");
```

Comment:
There's a few places that "queue" is hardcoded. Should it be added to NiFiTypes?

Reply:
Hmm, I think it should be ok as it is, since it's a 'name' property. It does not have to be consistent across different context, if it's 'typeName' or 'qualifiedName' or any other important attribute, then it probably should be a constant, but it's just a name. Thanks anyway.